src/server/components/formRunner/helpers.ts

```diff
@@ -128,12 +132,29 @@ export async function parseJsonFormData(
 
 interface initialiseFormRunnerInput {
   list: Pick<List, "type"> | Pick<ListItem, "type">;
-  listItem: ListItem;
+  listItem: ListItemWithHistory;
   message: string;
   isUnderTest: boolean;
   isAnnualReview?: boolean;
 }
 
+export function mergeUpdatedJson(listItem: ListItemWithHistory) {
+  const editEvent = listItem?.history.find((event) => {
+    // @ts-ignore
+    return event.type === ListItemEvent.EDITED && !!event.jsonData?.updatedJsonData;
+  });
```

Comment:
Edits from 1.55.x onwards will be on `updatedJsonData`. I'm going to check the prod database and see if there are any "old" style changes (via audit) outstanding. I'm tempted to leave them without this fix - and their non-cumulative updates can stay as they are


```.ts
// in listItemPutController

 const currentUpdatedJsonData = listItem.jsonData?.updatedJsonData ?? {}

 const newUpdates = data;
 
const newUpdatedJsonData = {...currentUpdatedJsonData, ...newUpdates };

```

```.ts
// in initialiseFormRunnerSession or mergeUpdatedJson

// this is destructuring using the "rest" property. 
// first it pulls out updatedJsonData, and the rest of the property/values will be assigned to a new const currentJsonData. 
const { updatedJsonData, ...currentJsonData } = listItem.jsonData;

const mergedJsonData = { ...currentJsonData, ...updatedJsonData }


```